src/Routing/ResourceResponse.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace DoubleThreeDigital\Runway\Routing;
+
+use Facades\Statamic\View\Cascade;
+use Illuminate\Contracts\Support\Responsable;
+use Illuminate\Support\Str;
+use Statamic\Events\ResponseCreated;
+use Statamic\Facades\Site;
+use Statamic\Statamic;
+use Statamic\View\View;
+
+class ResourceResponse implements Responsable
+{
+    protected $data;
+    protected $request;
+    protected $headers = [];
+    protected $with = [];
+
+    public function __construct($data)
+    {
+        $this->data = $data;
+    }
+
+    public function toResponse($request)
+    {
+        $this->request = $request;
+
+        $this
+            ->adjustResponseType()
+            ->addContentHeaders()
+            ->addViewPaths();
+
+        $response = response()
+            ->make($this->contents())
+            ->withHeaders($this->headers);
+
+        ResponseCreated::dispatch($response);
+
+        return $response;
+    }
+
+    protected function addViewPaths()
+    {
+        $finder = view()->getFinder();
+        $amp = Statamic::isAmpRequest();
+
+        $site = method_exists($this->data, 'site')
+            ? $this->data->site()->handle()
+            : Site::current()->handle();
+
+        $paths = collect($finder->getPaths())->flatMap(function ($path) use ($site, $amp) {
+            return [
+                $amp ? $path.'/'.$site.'/amp' : null,
+                $path.'/'.$site,
+                $amp ? $path.'/amp' : null,
+                $path,
+            ];
+        })->filter()->values()->all();
+
+        $finder->setPaths($paths);
+
+        return $this;
+    }
+
+    protected function contents()
+    {
+        $contents = (new View)
+            ->template($this->data->template())
+            ->layout($this->data->layout())
+            ->with($this->data->toAugmentedArray())
+            ->cascadeContent($this->data)
+            ->render();
+
+        if ($this->isLivePreview()) {
+            $contents = $this->versionJavascriptModules($contents);
+        }
+
+        return $contents;
+    }
+
+    protected function cascade()
+    {
+        return Cascade::instance()->withContent($this->data)->hydrate();
+    }
+
+    protected function adjustResponseType()
+    {
+        // $contentType = $this->data->get('content_type', 'html');
```

Comment:
Should we just get rid of this method altogether or make it work if a `content_type` field exists on the model?